### DISCUSSION TITLE
Correct way of rounding coordinates

### DISCUSSION THREAD

**Discussion by vkovenko** [pymupdf/PyMuPDF#2586] [Q&A]
Greetings. I'm looking for a way of correct rounding of coordinates from floats to integers. I need to do it as I'm working with svg elements and text bounding boxes parsed out from pdf using fitz. Can you provide some insights of what rounding logic is a correct one?
Thank you.

**Comment by JorjMcKie** [comment]
The canonic way is using `fitz.IRect`. Every rectangle `rect = fitz.Rect(x0, y0, x1, y1)` can be converted to its IRect via `rect.irect`.
The main point (plus some detail considerations) is that top and left are round downwards / to lower / floor integers), and right / bottom to the "ceiling" integers.

**Reply by vkovenko** [reply]
What about other objects like lines?

**Reply by JorjMcKie** [reply]
There is no definite, overall reason to ever do this. It fully depends on you specific app's requirements.
Like you may want to identify two lines if their defining points are close enough to each other or the like.
Same is true if you want to compute the line coordinates on the pixmap image of a page. There, all coordinates by their very nature are integers. Their is no rule for you how to do this rounding.
SVG images are in that sense no images, because they are no raster format - instead they smoothly zoom with the overall scaling. That is what makes them attractive after all.
